op-chain-ops/interopgen/deployers/opchain.go

```diff
@@ -0,0 +1,87 @@
+package deployers
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script"
+)
+
+type DeployOPChainInput struct {
+	OpChainProxyAdminOwner common.Address
+	SystemConfigOwner      common.Address
+	Batcher                common.Address
+	UnsafeBlockSigner      common.Address
+	Proposer               common.Address
+	Challenger             common.Address
+
+	BasefeeScalar     uint32
+	BlobBaseFeeScalar uint32
+	L2ChainId         *big.Int
+	Opsm              common.Address
+}
+
+func (input *DeployOPChainInput) InputSet() bool {
+	return true
+}
+
+type DeployOPChainOutput struct {
+	OpChainProxyAdmin                 common.Address
+	AddressManager                    common.Address
+	L1ERC721BridgeProxy               common.Address
+	SystemConfigProxy                 common.Address
+	OptimismMintableERC20FactoryProxy common.Address
+	L1StandardBridgeProxy             common.Address
+	L1CrossDomainMessengerProxy       common.Address
+	// Fault proof contracts below.
+	OptimismPortalProxy                common.Address
+	DisputeGameFactoryProxy            common.Address
+	DisputeGameFactoryImpl             common.Address
+	AnchorStateRegistryProxy           common.Address
+	AnchorStateRegistryImpl            common.Address
+	FaultDisputeGame                   common.Address
+	PermissionedDisputeGame            common.Address
+	DelayedWETHPermissionedGameProxy   common.Address
+	DelayedWETHPermissionlessGameProxy common.Address
```

Comment:
Any long-term value in making FP addresses a sub-structure? They take up more than half of these entries.

Reply:
We could, although they're all the same to the deploy script that creates the chain. Unless OPSM groups them more I am going to leave them like this for now.